### PR TITLE
finished user availability submission front end

### PR DESCRIPTION
**Pull request by wgwumass**

### CODE REVIEW

File: frontend/source/components/AvailabilityComponent/AvailabilityComponent.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                <label for="t${i}">${this.#potentialTimes[i].date}: ${this.#potentialTimes[i].startTime} - ${this.#potentialTimes[i].startTime}</label>
+            `;
+        }
+    }
+
+    #attachEventListeners() {
+        // Attach event listeners to the input and button elements
+        const submitTimesBtn = this.#container.querySelector("#availability-submit-button");
+        submitTimesBtn.addEventListener("click", () => this.#handleUserSubmission());
+    }
```
Comment (joshua041904) [inline frontend/source/components/AvailabilityComponent/AvailabilityComponent.js:83]: I think we have to connect this to EventCreationComponent. My EventCreationComponent publishes 2 events: "NewMeeting" and "StoreMeeting". The published data includes an array of invitees emails. I think maybe you have to subscribe to the NewMeeting event and call a function that sends out notifications to the emails.

File: frontend/source/components/AvailabilityComponent/AvailabilityComponent.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        this.#publishAvailableTimes(confirmedTimes);
+    }
+
+    #publishAvailableTimes(times) {
+        const hub = EventHub.getInstance();
+        hub.publish(Events.UserAvailabilitySubmission, times);
+    }
+
+    #sendInviteeNotification(meetingData) {
+        const invited = new Notification("you have been invited to " + meetingData[0]);
```
Comment (joshua041904) [inline frontend/source/components/AvailabilityComponent/AvailabilityComponent.js:105]: meetingData is an object. We may need to change meetingData[0] to meetingData.name

This is how I publised the event in EventCreation Component:
hub.publish(Events.NewMeeting, { name, description, invitees, times });
{ name, description, invitees, times } represents meeting data...name and description are strings, invitees and times are arrays,

File: frontend/source/components/AvailabilityComponent/AvailabilityComponent.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        return `
+            <div id="blue-header">
+                <h1>Submit Available Times</h1>
+            </div>
+
+            <div id="availability-input-body">
+                <h2>Event Details</h2>
+                <h3 id="eventName">${this.#eventName}</h3>
+
+                <h3 id="eventDescription">${this.#eventDescription}</h3>
```
Comment (joshua041904) [inline frontend/source/components/AvailabilityComponent/AvailabilityComponent.js:56]: I dont think the `${}` format works in HTML. Might have to reimplement this part with JS.

File: frontend/source/components/AvailabilityComponent/AvailabilityComponent.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        const timeList = this.#container.querySelector("#time-list");
+        for(let i = 0; i < this.#potentialTimes.length; i++) {
+            const timeContainer = document.createElement("div");
+            timeContainer.id = "time" + i;
+            timeContainer.innerHTML = `
+                <input type="checkbox" id="t${i}" class="time-input" name="t${i}" value="t${i}">
+                <label for="t${i}">${this.#potentialTimes[i].date}: ${this.#potentialTimes[i].startTime} - ${this.#potentialTimes[i].startTime}</label>
+            `;
+        }
+    }
```
Comment (joshua041904) [inline frontend/source/components/AvailabilityComponent/AvailabilityComponent.js:82]: perhaps need a line like this to add the timeList to the DOM:
timeList.appendChild(timeContainer);

File: frontend/source/components/AvailabilityComponent/AvailabilityComponent.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        const eventDescriptionElem = this.#container.querySelector("#eventDescription");
+        eventDescriptionElem.innerHTML = this.#eventDescription;
+        const timeList = this.#container.querySelector("#time-list");
+        for(let i = 0; i < this.#potentialTimes.length; i++) {
+            const timeContainer = document.createElement("div");
+            timeContainer.id = "time" + i;
+            timeContainer.innerHTML = `
+                <input type="checkbox" id="t${i}" class="time-input" name="t${i}" value="t${i}">
+                <label for="t${i}">${this.#potentialTimes[i].date}: ${this.#potentialTimes[i].startTime} - ${this.#potentialTimes[i].startTime}</label>
+            `;
```
Comment (joshua041904) [inline frontend/source/components/AvailabilityComponent/AvailabilityComponent.js:80]: you have startTime - startTime...is it supposed to be startTime - endTime?

**Review by LTan-101104 — APPROVED** [review]
Made some changes, it is good now

**Pull request merged**